tick/preprocessing/longitudinal_features_lagger.py

```diff
@@ -0,0 +1,250 @@
+
+
+class LongitudinalFeaturesLagger(Base):
+    """Transforms longitudinal exposure features to add columns representing 
+    lagged features. 
+
+    This preprocessor transform an input list of `n_samples` numpy ndarrays or 
+    scipy.sparse.csr_matrices of shape `(n_intervals, n_features)` so as to 
+    add columns representing the lagged exposures. It outputs a list of 
+    `n_samples` numpy arrays or  csr_matrices of shape 
+    `(n_intervals, n_features * (n_lags + 1))`.
+
+    Exposure can take two forms:
+    - short repeated exposures: in that case, each column of the numpy arrays 
+    or csr matrices can contain multiple ones, each one representing an exposure
+    for a particular time bucket.
+    - infinite unique exposures: in that case, each column of the numpy arrays
+    or csr matrices can only contain a single one, corresponding to the starting
+    date of the exposure.
+
+    Parameters
+    ----------
+    n_lags : `int`, default=0
+        Number of lags to compute: the preprocessor adds columns representing
+        lag = 1, ..., n_lags. If lag = 0, this preprocessor does nothing.
+    
+    Attributes
+    ----------
+    mapper : `dict`
+        Map lagged features to column indexes of the resulting matrices.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.sparse import csr_matrix
+    >>> from tick.preprocessing.longitudinal_features_lagger import LongitudinalFeaturesLagger
+    >>> features = [csr_matrix([[0, 1, 0],
+    ...                         [0, 0, 0],
+    ...                         [0, 0, 1]], dtype="float64"),
+    ...             csr_matrix([[1, 1, 0],
+    ...                         [0, 0, 1],
+    ...                         [0, 0, 0]], dtype="float64")
+    ...             ]
+    >>> censoring = np.array([3, 2], dtype="uint64").reshape((-1, 1))
+    >>> lfl = LongitudinalFeaturesLagger(n_lags=2)
+    >>> product_features = lfl.fit_transform(features)
+    >>> # output comes as a list of sparse matrices or 2D numpy arrays
+    >>> product_features.__class__
+    <class 'list'>
+    >>> [x.toarray() for x in product_features] 
+    [array([[ 0.,  0.,  0.,  1.,  0.,  0.,  0.,  0.,  0.],
+            [ 0.,  0.,  0.,  0.,  1.,  0.,  0.,  0.,  0.],
+            [ 0.,  0.,  0.,  0.,  0.,  1.,  1.,  0.,  0.]]),
+     array([[ 1.,  0.,  0.,  1.,  0.,  0.,  0.,  0.,  0.],
+            [ 0.,  1.,  0.,  0.,  1.,  0.,  1.,  0.,  0.],
+            [ 0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.,  0.]])]
+    """
+
+    _attrinfos = {
+        "n_lags": {"writable": False},
+        "_mapper": {"writable": False},
+        "_n_init_features": {"writable": False},
+        "_n_output_features": {"writable": False},
+        "_n_intervals": {"writable": False},
+        "_cpp_preprocessor": {"writable": False},
+        "_fitted": {"writable": False}
+    }
+
+    def __init__(self, n_lags=0, n_jobs=-1):
+        Base.__init__(self)
+        self.n_lags = n_lags
```

Comment:
Don't we need some tests about n_lags ? n_lags >= 0 and one checking that n_lags is not too large ?